src/Swarm/Language/Syntax.hs

```diff
@@ -1113,23 +1115,33 @@ freeVarsV = freeVarsT . (\f -> \case TVar x -> TVar <$> f x; t -> pure t)
 mapFreeS :: Var -> (Syntax' ty -> Syntax' ty) -> Syntax' ty -> Syntax' ty
 mapFreeS x f = freeVarsS %~ (\t -> case t ^. sTerm of TVar y | y == x -> f t; _ -> t)
 
+syntaxAsTree :: Syntax' a -> Tree (Syntax' a)
+syntaxAsTree s0@(Syntax' _ t _) = case t of
```

Comment:
I'll put the comment here too, to keep things in one place:
```
import Control.Lens.Plated (para)

syntaxAsTree :: Data a => Syntax' a -> Tree (Syntax' a)
syntaxAsTree = para Node
```

Reply:
:exploding_head: